js/main.js

```diff
@@ -35,3 +34,70 @@ const getRandomFloatingPointNumber = function (min, max, floatingPoint) {
 }
 
 getRandomFloatingPointNumber(1,2,1);
+
+// Module3-task1
+
+// Input data
+const TITLE = ['Информация о поездке', 'Описание предложения', 'Информация о месте проживания'];
+const ACCOMODATION_TYPE = ['palace', 'flat', 'house', 'bungalow'];
+const TIME = ['12:00', '13:00', '14:00'];
+const ACCOMMODATION_FACILITIES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+const PHOTOS = ['http://o0.github.io/assets/images/tokyo/hotel1.jpg', 'http://o0.github.io/assets/images/tokyo/hotel2.jpg', 'http://o0.github.io/assets/images/tokyo/hotel3.jpg'];
+const DESCRIPTION = ['Чисто, убрано, с крутым балконом', 'Большие номера с видом на море', 'Френдли к животным', 'Панорамные окна для фоточек']
+const ACCOMODATION_VARIANTS = 10;
+
+// Secondary functions
+function shuffle(array) {
+  array.sort(() => Math.random() - 0.5)
+}
+
+const getRandomAvatarNumber = (minNumber,maxNumber) => {
+  return ['0'+ getRandomInt(minNumber, maxNumber)].join();
+};
+
+const getRandomArrayIndex = (arr) => {
+  return arr[getRandomInt(0, arr.length - 1)]
+};
+
+const getRandomArray = (arr) => {
+  shuffle(arr);
+  const slicedArray = arr.slice(0, getRandomInt(1, arr.length - 1));
+  return slicedArray;
+};
+
+const getRandomFloatingPoints = (x, y, floatingPoint) => {
+  return getRandomFloatingPointNumber(x, y, floatingPoint);
+}
+
+// Result function
+const generateOffer = () => {
+
+  const locationX = getRandomFloatingPoints(35.65000, 35.70000, 5);
+  const locationY = getRandomFloatingPoints(139.70000, 139.80000, 5);
+
+  return {
+    'author': {
+      avatar: 'img/avatars/user' + getRandomAvatarNumber(1,8) + '.png',
+    },
+    'offer': {
+      title: getRandomArrayIndex(TITLE),
+      address: [locationX, locationY].toString(),
+      price: getRandomInt(0,5000),
+      type: getRandomArrayIndex(ACCOMODATION_TYPE),
+      rooms: getRandomInt(1,100),
+      guests: getRandomInt(1,100),
+      checkin: getRandomArrayIndex(TIME),
+      checkout: getRandomArrayIndex(TIME),
+      features: getRandomArray(ACCOMMODATION_FACILITIES),
+      description: getRandomArrayIndex(DESCRIPTION),
+      photos: getRandomArray(PHOTOS),
+    },
+    'location': {
+      x: locationX,
+      y: locationY,
+    },
+  };
+};
+
+const getRandomLocations = new Array(ACCOMODATION_VARIANTS).fill(null).map(() => generateOffer());
```

Comment:
можно не создавать функцию-обертку
```suggestion
const getRandomLocations = new Array(ACCOMODATION_VARIANTS).fill(null).map(generateOffer);
```